redisson/src/main/java/org/redisson/RedissonBaseLock.java

```diff
@@ -157,6 +163,8 @@ public void run(Timeout timeout) throws Exception {
                     if (res) {
                         // reschedule itself
                         renewExpiration();
+                    } else {
+                    	cancelExpirationRenewal(null);
```

Comment:
here is the key to solve the problem. Since it doesn't find a key to renew Expirations should cancel the expiration. Without this if this same instance tries to aquire a new lock it just add a new thread Key to the list and it doesn't have an active expiration renew flow.